src/types/onyx/OriginalMessage.ts

```diff
@@ -220,182 +130,169 @@ type ChronosOOOEvent = {
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_DELAYED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_QUEUED]: OriginalMessageReimbursementQueued;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_REQUESTED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_SETUP]: never;
+    [CONST.REPORT.ACTIONS.TYPE.RENAMED]: OriginalMessageRenamed;
+    [CONST.REPORT.ACTIONS.TYPE.REPORT_PREVIEW]: OriginalMessageReportPreview;
+    [CONST.REPORT.ACTIONS.TYPE.SELECTED_FOR_RANDOM_AUDIT]: never;
+    [CONST.REPORT.ACTIONS.TYPE.SHARE]: never;
+    [CONST.REPORT.ACTIONS.TYPE.STRIPE_PAID]: never;
+    [CONST.REPORT.ACTIONS.TYPE.SUBMITTED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TASK_CANCELLED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TASK_COMPLETED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TASK_EDITED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TASK_REOPENED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TAKE_CONTROL]: never;
+    [CONST.REPORT.ACTIONS.TYPE.UNAPPROVED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.UNHOLD]: never;
+    [CONST.REPORT.ACTIONS.TYPE.UNSHARE]: never;
+    [CONST.REPORT.ACTIONS.TYPE.UPDATE_GROUP_CHAT_MEMBER_ROLE]: never;
+} & {
+    [T in ValueOf<typeof CONST.REPORT.ACTIONS.TYPE.POLICY_CHANGE_LOG>]: OriginalMessageChangeLog;
+} & {
+    [T in ValueOf<typeof CONST.REPORT.ACTIONS.TYPE.ROOM_CHANGE_LOG>]: OriginalMessageChangeLog;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+type AssertOriginalMessageDefinedForAllActions = AssertTypesEqual<
+    ReportActionName,
+    keyof OriginalMessageMap,
+    `Error: Types don't match, OriginalMessageMap type is missing: ${Exclude<ReportActionName, keyof OriginalMessageMap>}`
+>;
 
-type OriginalMessage =
-    | OriginalMessageApproved
-    | OriginalMessageIOU
-    | OriginalMessageAddComment
-    | OriginalMessageActionableMentionWhisper
-    | OriginalMessageActionableReportMentionWhisper
-    | OriginalMessageSubmitted
-    | OriginalMessageClosed
-    | OriginalMessageCreated
-    | OriginalMessageHold
-    | OriginalMessageHoldComment
-    | OriginalMessageUnHold
-    | OriginalMessageRenamed
-    | OriginalMessageChronosOOOList
-    | OriginalMessageReportPreview
-    | OriginalMessageRoomChangeLog
-    | OriginalMessagePolicyChangeLog
-    | OriginalMessagePolicyTask
-    | OriginalMessageJoinPolicyChangeLog
-    | OriginalMessageModifiedExpense
-    | OriginalMessageReimbursementQueued
-    | OriginalMessageReimbursementDequeued
-    | OriginalMessageMoved
-    | OriginalMessageMarkedReimbursed
-    | OriginalMessageActionableTrackedExpenseWhisper
-    | OriginalMessageMergedWithCashTransaction
-    | OriginalMessageDismissedViolation;
+type OriginalMessage<T extends ReportActionName> = OriginalMessageMap[T];
+
+// Note: type-fest's ConditionalKeys does not work correctly with objects containing `never`: https://github.com/sindresorhus/type-fest/issues/878
+type ReportActionNamesWithHTMLMessage = {
```

Comment:
Note: I [fixed this upstream](https://github.com/sindresorhus/type-fest/pull/881), so if you upgrade type-fest to [>=4.18.3](https://github.com/sindresorhus/type-fest/releases/tag/v4.18.3), then you can just do this:

```ts
type ReportActionNamesWithHTMLMessage = ConditionalPick<OriginalMessageMap, {html: string}>;
```